test/src/nyc-test.js

```diff
@@ -232,6 +248,39 @@ describe('nyc', function () {
       })
     })
 
+    describe('compile handlers for custom extensions are assigned', function () {
+      it('assigns a function to custom extensions', function () {
+        var nyc = new NYC({
+          cwd: process.cwd()
```

Comment:
Better to provide `extension: '.es6'` here than rely on any `package.json` config. More explicit that way.